test/WireMock.Net.Tests/WireMockServerTests.cs

```diff
@@ -160,9 +160,8 @@ public async Task WireMockServer_Should_exclude_restrictedResponseHeader()
             Check.That(response.Headers.Contains("Transfer-Encoding")).IsFalse();
         }
 
-        [Theory]
-        [InlineData("DELETE")]
 #if !NET452
+        [Theory]
```

Comment:
I think this is not ok. Now the [Theory] is not defined for .net 452? This should result into a build-error? (BTW I cannot remember anymore why this is?)

Best to replace code like:
``` c#
[Theory]
[InlineData("PUT")]
[InlineData("OPTIONS")]
[InlineData("REPORT")]
[InlineData("DELETE")]
#if !NET452
[InlineData("TRACE")]
[InlineData("GET")]
#endif
```

Reply:
The issue is that I can't leave the [Theory] tag outside the if region since _that_ results in a build error because I can't have the [Theory] tag without an InlineData tag.

I'm building against 461 and I don't get any build error.

Reply:
@StefH I think this unit test is problematic at best. It looks like nothing inside the #if region is tested.
And once I took the tags out of the region & compiled using 461 (NOT 452), the unit test still failed.

I got an exception on `GET` as follows: `System.Net.ProtocolViolationException : Cannot send a content-body with this verb-type.`

For `TRACE`, the unit test failed - it returned a `BadRequest` instead of expected `OK`.

Maybe this unit test needs to be redesigned to not use `HttpRequestMessage` since we need to test a situation when a body actually arrives to the server.

Another possible solution would to allow body on all requests by using `System.Net.Http.WinHttpHandler` as [indicated here.](https://stackoverflow.com/a/47902348/10233008)

Reply:
What happens if you put exactly the code like:
``` c#
[Theory]
[InlineData("PUT")]
[InlineData("OPTIONS")]
[InlineData("REPORT")]
[InlineData("DELETE")]
#if !NET452
[InlineData("TRACE")]
[InlineData("GET")]
#endif
```

It should run all verbs for net-core and net46. And skip those 2 problems verbs for net452

Reply:
It fails, and ignores the 2 problem verbs even when using net461.

The failure here is because the 4 other verbs do allow the body, and then are caught by the first mapping which returns 400 BadRequest. This unit test specifically tests methods where body is not allowed, so including the 4 verbs where body is allowed will obviously fail the test.

Reply:
Sorry. Now I understand, there are 2 test methods...

OK.

**Fix 1:**
`WireMockServer_Should_exclude_body_for_methods_where_body_is_definitely_disallowed` should be completely surrounded by if-def NET452 in your code.

like:
``` c#
#if !NET452
        [Theory]
        [InlineData("TRACE")]
        [InlineData("GET")]
        public async Task WireMockServer_Should_exclude_body_for_methods_where_body_is_definitely_disallowed(string method)
        {
            // Assign
            ...
        }
#endif
```

**Fix 2:**
For test `WireMockServer_Should_not_exclude_body_for_supported_methods` : the "DELETE" should be added ( I think you already did this)

Reply:
Maybe I'm not building the project the correct way...
When these are my settings, everything inside the #if region does not show up in the test explorer... 

![image](https://user-images.githubusercontent.com/19075463/73851792-c643fe00-4836-11ea-9a61-3aa9a8665b0e.png)

Reply:
Maybe I did remove the net452 Target from the test project.

So if you fix my last two things, the PR is ready.

Reply:
Just pushed Fix 1. 
Fix 2 I already did as you mentioned.

Resolving this thread.